### ISSUE TITLE
Question about hidden states in onnx DistilGPT2

### ISSUE THREAD

**Issue by smolskayanastassia** [microsoft/onnxruntime#10382]
I tried to convert DistilGPT2 with this [notebook](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/transformers/notebooks/Inference_GPT2_with_OnnxRuntime_on_CPU.ipynb). 
I want to get hidden_states, past_key_values and logits.

Code:
```
from onnxruntime.transformers.gpt2_helper import Gpt2Helper, MyGPT2LMHeadModel
from transformers import AutoConfig
import torch
import os 

cache_dir = os.path.join(".", "cache_models")
model_name_or_path = "distilgpt2"
config = AutoConfig.from_pretrained(model_name_or_path)
config.output_hidden_states = True
config.output_attentions = True
config.output_scores = True
model = MyGPT2LMHeadModel.from_pretrained(model_name_or_path, config=config)
device = torch.device("cpu")
model.eval().to(device)

Gpt2Helper.export_onnx(model, device, onnx_model_path) 
```

I only got past_key_values and logits when I tested onnx model. Could you please tell how to get hidden_states?

**Comment by tianleiwu** [comment]
For "hidden_states", do you mean the attention weights mentioned in the following?
https://github.com/huggingface/transformers/blob/637e81752aad01738d36ef816fadee21fa392317/src/transformers/models/gpt2/modeling_gpt2.py#L343-L344

For that additional output in ONNX, you will need modify the source code here: https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/transformers/gpt2_helper.py
(1) modify MyGPT2LMHeadModel to get the extra output
(2) modify Gpt2Helper.export_onnx function to add output names, dynamic axes information etc.


Note that the exported ONNX model cannot be fully optimized since current Attention operator does not support that extra output right now. If you need optimization, that will need further modification in Attention operator and graph fusion logic.
